### ISSUE TITLE
Better Error for `invertedReturn->getType() == gutils->getShadowType(call.getType())`

### ISSUE THREAD

**Issue by vchuravy** [EnzymeAD/Enzyme.jl#1779]
```
vchuravy@loki ~/s/s/chmy_ad> julia --project=. example_2_ForwardDiff.jl 
 o:   %17 = call nonnull "enzyme_type"="{[-1]:Pointer}" {} addrspace(10)* ({} addrspace(10)* ({} addrspace(10)*, {} addrspace(10)**, i32)*, {} addrspace(10)*, ...) @julia.call({} addrspace(10)* ({} addrspace(10)*, {} addrspace(10)**, i32)* noundef nonnull @jl_f__svec_ref, {} addrspace(10)* noundef null, {} addrspace(10)* nonnull %16, {} addrspace(10)* addrspacecast ({}* inttoptr (i64 131314753880704 to {}*) to {} addrspace(10)*)) #17, !dbg !80
 ot: {} addrspace(10)*
 ir:   %424 = call nonnull "enzyme_type"="{[-1]:Pointer}" {} addrspace(10)* ({} addrspace(10)* ({} addrspace(10)*, {} addrspace(10)**, i32)*, {} addrspace(10)*, ...) @julia.call({} addrspace(10)* ({} addrspace(10)*, {} addrspace(10)**, i32)* noundef nonnull @jl_f__svec_ref, {} addrspace(10)* noundef null, {} addrspace(10)* nonnull %415, {} addrspace(10)* addrspacecast ({}* inttoptr (i64 131314753880704 to {}*) to {} addrspace(10)*)) #17, !dbg !486
 irt: {} addrspace(10)*
 p:   %"'dual_phi2" = phi [132 x {} addrspace(10)*] , !dbg !486
 PT: [132 x {} addrspace(10)*]
 newCall:   %424 = call nonnull "enzyme_type"="{[-1]:Pointer}" {} addrspace(10)* ({} addrspace(10)* ({} addrspace(10)*, {} addrspace(10)**, i32)*, {} addrspace(10)*, ...) @julia.call({} addrspace(10)* ({} addrspace(10)*, {} addrspace(10)**, i32)* noundef nonnull @jl_f__svec_ref, {} addrspace(10)* noundef null, {} addrspace(10)* nonnull %415, {} addrspace(10)* addrspacecast ({}* inttoptr (i64 131314753880704 to {}*) to {} addrspace(10)*)) #17, !dbg !486
 newCallT: {} addrspace(10)*
julia: /workspace/srcdir/Enzyme/enzyme/Enzyme/AdjointGenerator.h:6174: void AdjointGenerator::visitCallInst(llvm::CallInst&): Assertion `invertedReturn->getType() == gutils->getShadowType(call.getType())' failed.

```

**Comment by wsmoses** [comment]
can you attach a MWE? generally that is an issue that Enzyme.jl should prevent from happening

**Comment by vchuravy** [comment]
Can confirm fixed for me locally after #1782

**Comment by vchuravy** [comment]
The reproducer was `Enzyme.jacobian(Forward, (x)->(Res_closed!(x, Cvec);Cvec), Rvec)` with the code from  #1781
